posydon/popsyn/GRB.py

```diff
@@ -0,0 +1,119 @@
+__author__ = ['Simone Bavera <Simone.Bavera@unige.ch>']
+
+import warnings
+import numpy as np
+from posydon.utils.constants import Msun, clight
+
+GRB_PROPERTIES = ['GRB1', 'S1_eta', 'S1_f_beaming', 'S1_E_GRB', 'S1_E_GRB_iso', 'S1_L_GRB_iso',
+                  'GRB2', 'S2_eta', 'S2_f_beaming', 'S2_E_GRB', 'S2_E_GRB_iso', 'S2_L_GRB_iso']
+
+def get_GRB_properties(df, GRB_efficiency, GRB_beaming, E_GRB_iso_min=0.):
+    """Compute GRB properties for a given binary population."""
+    
+    # check if radiated disk mass is avaialble to compute GRB properties
+    if ('S1_m_disk_radiated' not in df or
+        'S2_m_disk_radiated' not in df):
+        raise ValueError('m_disk_radiated not found in the dataframe!')
+    
+    # reminder the user about the threshold cut
+    if E_GRB_iso_min != 0.:
+        warnings.warn("We only consider GRBs with isotropic equivalent "
+                        f"energy larger than {E_GRB_iso_min} erg!")
+        
+    # define some methods
+    
+    def compute_eta(i, sel):
+        """Compute GRB energy conversion efficiency factor."""
+        if (not isinstance(GRB_efficiency, float) and
+            GRB_efficiency >= 0. and GRB_efficiency <= 1.):
```

Comment:
It looks like a bracket is missing here, it should read `(not (isinstance(GRB_efficiency, float) and GRB_efficiency >= 0. and GRB_efficiency <= 1.))` or `(not isinstance(GRB_efficiency, float) or GRB_efficiency < 0. or GRB_efficiency > 1.)`

Reply:
In Python, if the first condition is `False` the other two `and` conditions will never be executed as the logic directly evaluates to `False`. So in this case the first condition will check for the variable not being a float, if it is a float it will continue to evaluate the other two, one by one.

Reply:
The point here was not about the order, it is about the logic:
1. if it is not a float
2. if it is larger or equal to 0
3. if it is smaller or equal to 1
We should get the error, when 1. is not fulfilled, and when it is not in the range [0,1].
So here a third way to write that: not isinstance(GRB_efficiency, float) or not (GRB_efficiency >= 0. and GRB_efficiency <= 1.)

Reply:
done